### PR TITLE
Refactor/goal progress

### PR DESCRIPTION
**Pull request by marcianosr**
## What

Explain what changes inside the code, this can be as simple or complicated as you want as long as it's clear

## Why

Explain why these things are changes. This explanation is for your colleagues and your future self.

## Code Review

Please consider the following checklist when reviewing this Pull Request.  
More background and details [here](https://github.com/kabisa/kabisa-guide/blob/master/organization/process/code-review/README.md).

* [ ] Does the code **actually solve** the problem it was meant to solve?
* [ ] Is the code covered by **unit tests**? **Integration tests**?
* [ ] Does anything here need **documentation**? (Focus on *why*, not *what.*)
* [ ] Does any of this code deal with **privacy sensitive information** or affects **security**? Ask an additional reviewer.
* [ ] Is the code easy to **understand** and **change** in the future?
* [ ] Is the same code or concept **duplicated**? Find a balance between DRYness and readability.
* [ ] Does the code reasonably adhere to the Kabisa [**coding standards**](https://github.com/kabisa/kabisa-guide/blob/master/organization/process/code-review/README.md)?
* [ ] Be kind.

### CODE REVIEW

File: src/ui/Progress/GoalProgressIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+type GoalProgressIndicatorProps = {
+  goals: Goal[];
+  activeKudosMeter: ActiveGoalAmount;
+};
+
+const GoalProgressIndicator = ({
+  goals,
+  activeKudosMeter,
+}: GoalProgressIndicatorProps) => {
+  const goalsDisplay = [...goals].reverse();
```
Comment (sandercamp) [inline src/ui/Progress/GoalProgressIndicator.tsx:25]: This depends on the assumption that the goals passed into this component are sorted correctly (low to high). I'm not sure if that is the case. I think the API returns an array of Goal objects sorted by id.

File: src/ui/Progress/GoalProgressIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+const Step = ({ goal }: StepProps) => (
+  <div className={styles.stepsContainer}>
+    <div
+      className={classNames(styles.step, {
+        [styles.stepEmpty]: !goal.achievedOn,
+        [styles.stepAchieved]: goal.achievedOn,
+      })}
+    >
+      {!goal.achievedOn && <Icon name="lock" className={styles.stepIcon} />}
```
Comment (sandercamp) [inline src/ui/Progress/GoalProgressIndicator.tsx:82]: You could reduce coupling with the Goal object by using a plain boolean prop that determines if this step is achieved or not

**Review by sandercamp — CHANGES_REQUESTED** [review]

File: src/ui/Progress/GoalProgressIndicator.tsx (hunk 1 above)
Comment (marcianosr) [reply src/ui/Progress/GoalProgressIndicator.tsx:25]: Now that you say it, I'm not entirely sure iether anymore. Will check!

File: src/ui/Progress/GoalProgressIndicator.tsx (hunk 2 above)
Comment (marcianosr) [reply src/ui/Progress/GoalProgressIndicator.tsx:82]: Good one!

**Review by sandercamp — APPROVED** [review]

**Pull request merged**